optimum/onnxruntime/base.py

```diff
@@ -288,6 +304,24 @@ def compute_past_key_values_output_shapes(
 
         return {name: key_shape if "key" in name else value_shape for name in self.key_value_output_names}
 
+    def compute_past_key_values_output_shapes_mqa(
+        self,
+        input_ids: torch.Tensor,
+        use_cache_branch: Optional[bool],
+        past_key_values: Optional[Tuple[Tuple[torch.FloatTensor]]] = None,
+    ) -> Dict[str, List[int]]:
+        batch_size = input_ids.size(0)
+        num_attention_heads = self.normalized_config.num_attention_heads
+        embed_size_per_head = self.normalized_config.hidden_size // num_attention_heads
+
+        sequence_length = input_ids.size(1)
+        if past_key_values is not None and use_cache_branch is not False:
+            sequence_length += past_key_values[0].size(-2)
```

Comment:
This line gives me the error:

`tuple object has no attribute size`. This is expected because as the typing in the function signature says, `past_key_values` is a tuple of tuples. Hence, `past_key_values[0]` is a tuple as well, not a `torch.Tensor`.

You probably copy pasted this signature and the code from the existing function `compute_past_key_values_output_shapes` but before calling that function we flatten `past_key_values` [here](https://github.com/huggingface/optimum/pull/1042/files#diff-b2296ed4846e14e948034c77f89f02dce3eabf97cddb47eac16b29ada232a06dR337). Therefore, the parameter type there is tuple of tensors. Note that the typing in function signature is wrong, because it says `Tuple[Tuple[Tensor]`. 

I think the correct code here should be: 

`sequence_length += past_key_values[0][0].size(-2)`

What do you say?